pkg/controller/deployment/unsupported_config_overrides_test.go

```diff
@@ -14,7 +14,12 @@ func TestUnsupportedConfigOverrides(t *testing.T) {
 	deploymentFile, err := assets.Asset("cert-manager-deployment/controller/cert-manager-deployment.yaml")
 	require.NoError(t, err)
 	testDeployment := resourceread.ReadDeploymentV1OrDie(deploymentFile)
-	defaultArgs := testDeployment.Spec.Template.Spec.Containers[0].Args
+
+	defaultArgs := make([]string, len(testDeployment.Spec.Template.Spec.Containers[0].Args))
+	copy(defaultArgs, testDeployment.Spec.Template.Spec.Containers[0].Args)
+
+	testArgs := []string{"--test-arg=true"}
```

Comment:
Can you also cover other variations of args?
```
--test=a
--test
--test=a=b=c
```

Reply:
You can cover this part in a separate unit test to cover `parseArgMap`/`updateArgMap`. Up to you.

Reply:
Added a test for `parseArgMap` with more extensive coverage (including similar cases as described).